pkg/reconciler/openshift/openshift.go

```diff
@@ -1,17 +1,20 @@
 package openshift
 
 import (
+	"log"
 	"os"
 	"strings"
 
 	argoprojv1alpha1 "github.com/argoproj-labs/argocd-operator/pkg/apis/argoproj/v1alpha1"
-	rbacv1 "k8s.io/api/rbac/v1"
-
 	"github.com/argoproj-labs/argocd-operator/pkg/controller/argocd"
+	rbacv1 "k8s.io/api/rbac/v1"
 )
 
 func init() {
 	argocd.Register(reconcilerHook)
+	if err := argocd.VerifyRouteAPI(); err != nil {
+		log.Fatal(err)
+	}
 }
 
```

Comment:
@bigkevmcd This module should really live downstream.   As long as `argocd.Register() enables a way to registry hooks.   It is downstream's job to register reconciler hooks.  They are not something that upstream cares about or needs to maintain.

Reply:
@wtam2018 I don't disagree, the separation between "gitops operator" and "argocd-operator" is not very clear, and if we're not planning on releasing the "argocd-operator" we should be clearer in our separation.